src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -703,58 +704,32 @@ private BoundStatement BindDeclarationStatementParts(LocalDeclarationStatementSy
-            // PROTOTYPE: what happens if there is a no arg int returning method, but a params object[] void returning. Currently, overload resolution would choose the int return, and we'd fail here.
-            if (disposeMethod is null || !disposeMethod.ReturnsVoid)
-            {
-                lookupResult.Clear();
-
-                var extDisposeMethod = FindPatternExtensionMethod(typeExpr,
-                                                                  WellKnownMemberNames.DisposeMethodName,
-                                                                  syntaxNode,
-                                                                  diagnostics,
-                                                                  MessageID.IDS_Disposable);
-
-                // if we still can't satisfy the lookup via extension method, report the diagnostics from both lookups
-                if (extDisposeMethod is null || !extDisposeMethod.ReturnsVoid)
-                {
-                    diagnostics.AddRange(initialDiagnostics);
-                    if (disposeMethod?.ReturnsVoid == false)
-                    {
-                        ReportPatternWarning(diagnostics, typeExpr.Type, disposeMethod, syntaxNode, MessageID.IDS_Disposable);
-                    }
-                    if (extDisposeMethod?.ReturnsVoid == false)
-                    {
-                        ReportPatternWarning(diagnostics, typeExpr.Type, extDisposeMethod, syntaxNode, MessageID.IDS_Disposable);
-                    }
-                    extDisposeMethod = null;
-                }
+        /// <param name="expr">Expression on which to perform lookup</param>
+        /// <param name="syntaxNode">The syntax node to perform lookup on</param>
+        /// <param name="diagnostics">Populated with invocation errors, and warnings of near misses</param>
+        /// <returns>The <see cref="MethodSymbol"/> of the Dispose method if one is found, otherwise null.</returns>
+        internal MethodSymbol TryFindDisposePatternMethod(BoundExpression expr, SyntaxNode syntaxNode, DiagnosticBag diagnostics)
+        {
+            if(expr?.Type is null)
+            {
+                return null;
+            }
 
-                disposeMethod = extDisposeMethod;
+            var result = FindPatternMethodRelaxed(expr,
+                                                  WellKnownMemberNames.DisposeMethodName,
+                                                  syntaxNode,
+                                                  diagnostics,
+                                                  out var disposeMethod);
+
+            
+            if (disposeMethod?.ReturnsVoid == false || result == PatternLookupResult.NotAMethod)
+            {
+                ReportPatternWarning(diagnostics, expr.Type, disposeMethod, syntaxNode, MessageID.IDS_Disposable);
+                disposeMethod = null;
             }
```

Comment:
>} [](start = 12, length = 1)

For binding `fixed` pattern, we do some extra checks (copied below). Just want to confirm that we're intentionally not doing those checks for `using` pattern.
It would be good to have a test for `static` `Dispose` method (that's not an extension).

```C#
if (HasOptionalOrVariableParameters(patterMethodSymbol) ||
                    patterMethodSymbol.ReturnsVoid ||
                    !patterMethodSymbol.RefKind.IsManagedReference() ||
                    !(patterMethodSymbol.ParameterCount == 0 || patterMethodSymbol.IsStatic && patterMethodSymbol.ParameterCount == 1))
                {
                    // the method does not fit the pattern
                    additionalDiagnostics.Add(ErrorCode.WRN_PatternBadSignature, initializer.Syntax.Location, initializer.Type, "fixed", patterMethodSymbol);
                    return null;
                }
```

Reply:
Generalized pattern lookup has two phases: lookup and verification. We're sharing the lookup phases, but each pattern has its own unique verification. In this case dispose is a bit more lenient than fixed. 

Added a test for the static method case.